solutions/USACO open19-milk-factory.mdx

```diff
@@ -153,3 +155,57 @@ int main(){
 
 </CPPSection>
 </LanguageSection>
+
+## Solution 2
+
+This solution follows the logic outlined in the official solution.
+
+## Implementation
+
+**Time Complexity**: $\mathcal{O}(N)$
+
+<LanguageSection>
+<PySection>
+
+```py
+# Take in the input using standard Python file i/o system
+data = open('factory.in').read().strip().split('\n')
+n = int(data[0])
+conveyer_belts = [list(map(int, data[i + 1].split(' '))) for i in range(n - 1)]
+
+# Create two lists holding all stations with incoming conveyer belts
+# Create another list holding all stations with outgoing conveyer belts
+outgoing = []
+incoming = []
```

Comment:
i don't think you need these? just keep track of the number of `outgoing_locations` for each station